src/ol/layer/group.js

```diff
@@ -221,6 +221,10 @@ ol.layer.Group.prototype.getLayerStatesArray = function(opt_states) {
         layerState.extent = ownLayerState.extent;
       }
     }
+    if (layerState.layer.getZIndex() === undefined) {
+      layerState.zIndex = ownLayerState.zIndex;
+      layerState.layer.setZIndex(ownLayerState.zIndex);
```

Comment:
Do you need to set the zIndex here?
I would expect that the internal mechanism of computing default zIndex would not interfere with the values set by the user.